### PR TITLE
Update user documentation

### PR DESCRIPTION
**Pull request by cshanahan1**
Overview of changes:

added overview of package and logo to landing page

- made a `Getting Started` section above `User Documentation`
      - Added a Quickstart guide to this section.
      -  Below that, detailed install instructions and contribution guide,  both of which have their own RST page but link to the github .md files (figured it was best to have this in one place so they don’t become out of sync, and they’re pretty thorough there).

- expanded and sectioned the sidebar to easier find pages

- expand toctrees on landing page to maxdepth == 2. In my personal opinion, this makes things much easier to find (especially in package index)

- reorganize and expand User Documentation section. 
	- Begins with an intro to the pipeline, then an overview of the pipeline code.
	- Sections each for general topics relevant to both strun and Python interface
		 * Reference files / CRDS
		 * Parameters
		 * Inputs and outputs
		 * Available pipeline modules, datamodels.
	- A section each for running/configuring the pipeline w/ strun and w/ Python interface. Here I repeated/rephrased some of the information already in the stpipe documentation, but I think it makes it easier to find and also provides some concrete examples for configuration with both interfaces. I tried to keep the Python and strun sections as similar as possible, showing how to do the same things with both interfaces.

### CODE REVIEW

**Comment by zacharyburnett** [comment]
It looks like the following warnings are being flagged by RTD before the build proceeds:
```
/home/docs/checkouts/readthedocs.org/user_builds/jwst-pipeline/checkouts/6919/docs/getting_started/faq.rst:13: WARNING: Inline interpreted text or phrase reference start-string without end-string.
/home/docs/checkouts/readthedocs.org/user_builds/jwst-pipeline/checkouts/6919/docs/getting_started/faq.rst:32: WARNING: Title underline too short.

How do I install an older version of the pipeline?
------------------------------------------------
/home/docs/checkouts/readthedocs.org/user_builds/jwst-pipeline/checkouts/6919/docs/getting_started/faq.rst:32: WARNING: Title underline too short.

How do I install an older version of the pipeline?
------------------------------------------------
/home/docs/checkouts/readthedocs.org/user_builds/jwst-pipeline/checkouts/6919/docs/getting_started/faq.rst:41: ERROR: Unexpected indentation.
/home/docs/checkouts/readthedocs.org/user_builds/jwst-pipeline/checkouts/6919/docs/getting_started/faq.rst:42: WARNING: Block quote ends without a blank line; unexpected unindent.
/home/docs/checkouts/readthedocs.org/user_builds/jwst-pipeline/checkouts/6919/docs/getting_started/faq.rst:136: WARNING: Title underline too short.

Can I create my own pipeline? # note to reviewer, is this answer right?
-----------------------------
/home/docs/checkouts/readthedocs.org/user_builds/jwst-pipeline/checkouts/6919/docs/getting_started/install.rst:47: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/docs/checkouts/readthedocs.org/user_builds/jwst-pipeline/checkouts/6919/docs/getting_started/install.rst:64: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/docs/checkouts/readthedocs.org/user_builds/jwst-pipeline/checkouts/6919/docs/getting_started/install.rst:75: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/docs/checkouts/readthedocs.org/user_builds/jwst-pipeline/checkouts/6919/docs/getting_started/install.rst:85: ERROR: Unexpected indentation.
/home/docs/checkouts/readthedocs.org/user_builds/jwst-pipeline/checkouts/6919/docs/jwst/user_documentation/reference_files_crds.rst:10: WARNING: Definition list ends without a blank line; unexpected unindent.
/home/docs/checkouts/readthedocs.org/user_builds/jwst-pipeline/checkouts/6919/docs/jwst/user_documentation/reference_files_crds.rst:52: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/docs/checkouts/readthedocs.org/user_builds/jwst-pipeline/checkouts/6919/docs/jwst/user_documentation/running_pipeline_command_line.rst:94: CRITICAL: Title level inconsistent:

Output Directory
^^^^^^^^^^^^^^^^
/home/docs/checkouts/readthedocs.org/user_builds/jwst-pipeline/checkouts/6919/docs/jwst/user_documentation/running_pipeline_command_line.rst:119: CRITICAL: Title level inconsistent:

Output File
^^^^^^^^^^^
/home/docs/checkouts/readthedocs.org/user_builds/jwst-pipeline/checkouts/6919/docs/jwst/user_documentation/running_pipeline_command_line.rst:159: CRITICAL: Title level inconsistent:

Override Reference File
^^^^^^^^^^^^^^^^^^^^^^^
/home/docs/checkouts/readthedocs.org/user_builds/jwst-pipeline/checkouts/6919/docs/jwst/user_documentation/running_pipeline_command_line.rst:178: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/docs/checkouts/readthedocs.org/user_builds/jwst-pipeline/checkouts/6919/docs/jwst/user_documentation/running_pipeline_command_line.rst:178: CRITICAL: Title level inconsistent:

Skip
^^^^
/home/docs/checkouts/readthedocs.org/user_builds/jwst-pipeline/checkouts/6919/docs/jwst/user_documentation/running_pipeline_python.rst:162: WARNING: Explicit markup ends without a blank line; unexpected unindent.
/home/docs/checkouts/readthedocs.org/user_builds/jwst-pipeline/checkouts/6919/docs/jwst/user_documentation/running_pipeline_python.rst:185: ERROR: Unexpected indentation.
/home/docs/checkouts/readthedocs.org/user_builds/jwst-pipeline/checkouts/6919/docs/jwst/user_documentation/running_pipeline_python.rst:189: WARNING: Explicit markup ends without a blank line; unexpected unindent.
```

File: docs/getting_started/faq.rst [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+Installation and Setup
+=======================
+
+How do I install the pipeline?
+------------------------------
+
+The most simple way to install the pipeline is to obtain the latest released
+version via. pip and install this into a Python environment. See the
+:ref:`QuickStart guide<quickstart>`for basic installation instructions.
```
Comment (zacharyburnett) [inline docs/getting_started/faq.rst:15]:
```suggestion
:ref:`QuickStart guide<quickstart>` for basic installation instructions.
```

File: docs/getting_started/faq.rst [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+It is also possible to install older released versions, or the development version
+from Github. For more detailed install instructions, see :ref:`here<installation>`.
```
Comment (zacharyburnett) [inline docs/getting_started/faq.rst:18]:
```suggestion
It is also possible to :ref:`install older released versions, or the development version from Github<installation>`.
```

File: docs/getting_started/faq.rst [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+The absolute most up-to-date version of `jwst` is the master branch of the Github
+repository. Changes are merged here on a regular basis, and this branch is
+constantly evolving between releases. If, for example, a bug is fixed
+in the pipeline and you would like to use this fixed version now rather than
+waiting for the next release, you will want to install this development version.
+See see :ref:`Installing the Development Version from Github<installing_dev>`
+for instructions on how to do this.
+
+How do I install an older version of the pipeline?
+------------------------------------------------
```
Comment (zacharyburnett) [inline docs/getting_started/faq.rst:32]:
```suggestion
--------------------------------------------------
```

File: docs/getting_started/faq.rst [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+
+See :ref:`python_outputs`. 
+
+
+General
+=======
+
+Can I create my own pipeline? # note to reviewer, is this answer right?
+-----------------------------
+
```
Comment (zacharyburnett) [inline docs/getting_started/faq.rst:142]:
```suggestion
Can I create my own pipeline?
-----------------------------

..
    # note to reviewer, is this answer right?
```

File: docs/getting_started/install.rst [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    | >> pip install jwst==1.3.3
+
+Installing specific versions before `jwst 0.17.0` need to be installed from Github:
+
+    | >> conda create -n <env_name> python
+    | >> conda activate <env_name>
+    | >> pip install git+https://github.com/spacetelescope/jwst@0.16.2
+
+
+.. _installing_dev:
```
Comment (zacharyburnett) [inline docs/getting_started/install.rst:63]:
```suggestion
.. _installing_dev:

```

File: docs/getting_started/install.rst [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+----------------------------------------------
+
+You can install the latest development version (not as well tested) from the
+Github master branch:
+
+    | >> conda create -n <env_name> python
+    | >> conda activate <env_name>
+    | >> pip install git+https://github.com/spacetelescope/jwst
+
+.. _upgrade_install:
```
Comment (zacharyburnett) [inline docs/getting_started/install.rst:74]:
```suggestion
.. _upgrade_install:

```

File: docs/jwst/user_documentation/running_pipeline_python.rst [hunk 7]
```diff
@@ -0,0 +1,335 @@
...
+can change the ones you desire.
+
+In short, **``call``** is the recommended way to use the pipeline and it uses parameter
+files to direct processing, while ``run`` requires you to do all that set up yourself.
+
+.. _python_outputs:
+Controlling Output Behavior
+---------------------------
+
+.. _python_output_file:
```
Comment (zacharyburnett) [inline docs/jwst/user_documentation/running_pipeline_python.rst:249]:
```suggestion
.. _python_output_file:

```

**Review by zacharyburnett — CHANGES_REQUESTED** [review]

File: docs/jwst/user_documentation/running_pipeline_python.rst [hunk 8]
```diff
@@ -0,0 +1,379 @@
+.. _run_from_python:
+
+===========================================
+Running the JWST pipeline: Python Interface
+===========================================
+
+.. note:: The use of the ``run`` method to run a pipeline or step is not
```
Comment (zacharyburnett) [inline docs/jwst/user_documentation/running_pipeline_python.rst:7]:
```suggestion
.. note:: 
    The use of the ``run`` method to run a pipeline or step is not
```

**Review by zacharyburnett — DISMISSED** [review]

File: docs/getting_started/faq.rst [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+
+General
+=======
+
+Can I create my own pipeline?
+-----------------------------
+
+..
+    # note to reviewer, is this answer right?
```
Comment (zacharyburnett) [inline docs/getting_started/faq.rst:144]:
```suggestion
    # note to reviewer, is this answer right?

```

**Comment by zacharyburnett** [comment]
@cshanahan1 it is giving a warning about `skip_step_cli`, but I can't seem to find that anchor anywhere. Is it supposed to be in the `faq.rst` file?
```
/home/docs/checkouts/readthedocs.org/user_builds/jwst-pipeline/checkouts/6919/docs/getting_started/faq.rst:128: WARNING: undefined label: 'skip_step_cli'
```

**Comment by cshanahan1** [comment]
> @cshanahan1 it is giving a warning about `skip_step_cli`, but I can't seem to find that anchor anywhere. Is it supposed to be in the `faq.rst` file?
> 
> ```
> /home/docs/checkouts/readthedocs.org/user_builds/jwst-pipeline/checkouts/6919/docs/getting_started/faq.rst:128: WARNING: undefined label: 'skip_step_cli'
> ```

i am working on that file right now and haven't commited it yet, stay tuned

**Review by nden — COMMENTED** [review]
Initial pass, just marking some typos.

File: docs/getting_started/install.rst [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+This will ensure that all dependency packages are also upgraded. This also
+applies when using the development version of jwst - to upgrade and grab recent
+changes, uninstall and re-install the master branch from Github:
+
+    | >> pip uninstall jwst
+    | >> pip install git+https://github.com/spacetelescope/jwst
+
+
+**For more install instructions, including how to install jwst for development**
+**or how to install a DMS operational build, see** `the Github README <https://github.com/spacetelescope/jwst>`_.
```
[the remark this answers is not in the document]
Comment (cshanahan1) [reply docs/getting_started/install.rst:112]: i added a small reference to this

**Review by nden — APPROVED** [review]
LGTM. Finally the docs mention JWST on the front page.

File: docs/jwst/user_documentation/running_pipeline_python.rst [hunk 11]
```diff
@@ -0,0 +1,528 @@
...
+
+Another option for running the pipeline is to use the `.run()` method instead
+of the `.call()` method. This method is not generally reccomended and considered
+advanced use, but it is an option to users.
+
+There is additional internal setup done when using the ``.call()`` method to ensure
+that reference files and parameter files from CRDS are passed to the pipeline along
+with any other parameters passed directly to the method. In the process of doing this,
+a new instance of the pipeline/step class is created internally, to which the final
+set of parameters is passed to and the ``.run()`` method, which is what finally begins
```
Comment (hbushouse) [inline docs/jwst/user_documentation/running_pipeline_python.rst:504]: I'm not sure how to parse this whole sentence. Are you saying that the way it works *internally* when you use .call() is that a new instance is created, parameters are retrieved, and then a call is made to .run() using those parameters? If so, I think the sentence needs a little rearranging to make that more obvious.

File: docs/jwst/user_documentation/running_pipeline_python.rst [hunk 12]
```diff
@@ -0,0 +1,528 @@
...
+processing, is invoked on this new internal instance. Attributes set on the original
+instance on which ``.call()`` is invoked won't be used during processing, they must be
```
Comment (hbushouse) [inline docs/jwst/user_documentation/running_pipeline_python.rst:506]: should be "... during processing. They must be ..."  (i.e. separate sentences)

File: docs/jwst/user_documentation/running_pipeline_python.rst [hunk 13]
```diff
@@ -0,0 +1,528 @@
...
+passed in as arguments to ``.call()``.
+
+In contrast, when using the ``.run()`` method directly on a pipeline/step, the
+additional logic to determine parameters and reference files is skipped. The pipeline
+instance is being run as-is, and coded defaults for the pipeline and each intermediate step
+will be used unless explicitley overridden individually. Because the instance created is
```
Comment (hbushouse) [inline docs/jwst/user_documentation/running_pipeline_python.rst:512]: typo "explicitley"

File: docs/jwst/user_documentation/running_pipeline_python.rst [hunk 14]
```diff
@@ -0,0 +1,528 @@
...
+being ran directly on the data, attributes can be set directly:
```
Comment (hbushouse) [inline docs/jwst/user_documentation/running_pipeline_python.rst:513]: typo: ran -> run

File: docs/jwst/user_documentation/running_pipeline_python.rst [hunk 15]
```diff
@@ -0,0 +1,528 @@
...
+::
+
+	from jwst.pipeline import Detector1Pipeline
+	pipe = Detector1Pipeline()
+	pipe.jump.threshold = 12
+	pipe.ramp_fit.skip = True
+	result = pipe.run('jw00017001001_01101_00001_nrca1_uncal.fits')
+
+The ``pipe`` object created and the attributes set will persist and this object
+can be resued within a Python session for processing data. Keep in mind that each
```
Comment (hbushouse) [inline docs/jwst/user_documentation/running_pipeline_python.rst:524]: typo: resused -> reused

**Review by hbushouse — COMMENTED** [review]
Phew! That was a lot to review!  See many comments.

File: docs/jwst/user_documentation/input_output_file_conventions.rst [hunk 16]
```diff
@@ -0,0 +1,77 @@
...
+- The name of the input file
+- The product name defined in an association
+- As specified by the :ref:`output_file <intro_output_file>` parameter
+
+Regardless of the source, each pipeline/step uses the name as a base
+name, onto which several different suffixes are appended, which
+indicate the type of data in that particular file. A list of the main suffixes
+can be :ref:`found below <pipeline_step_suffix_definitions>`.
+
+The pipelines do not manage versions. When re-running a pipeline, previous files
```
[the remark this answers is not in the document]
Comment (cshanahan1) [reply docs/jwst/user_documentation/input_output_file_conventions.rst:41]: this text was already here and im not sure what it means in this context to be honest. i'll change it to that though.

**Comment by nden** [comment]
This needs a rebase again

File: docs/jwst/user_documentation/reference_files_crds.rst [hunk 17]
```diff
@@ -0,0 +1,155 @@
...
+Parameter Files
+===============
+
+Parameter files, which like reference files are encoded in ASDF and
+version-controlled by CRDS, define the 'best' set of parameters for pipeline
+steps as determined by the JWST instrument teams, based on insturment, observing
+model, filter, etc. They also may evolve over time as understanding of caibration
+improves.
+
+By default, when running the pipeline via ``strun`` or using the ``pipeline/step.run()``
```
[the remark this answers is not in the document]
Comment (cshanahan1) [reply docs/jwst/user_documentation/reference_files_crds.rst:46]: whoops, thats what i meant.

File: docs/jwst/user_documentation/reference_files_crds.rst [hunk 18]
```diff
@@ -0,0 +1,155 @@
...
+
+  $ export CRDS_CONTEXT='jwst_0421.pmap'
+
+For all information about CRDS, including context lists, see the JWST CRDS
+website:
+
+    `https://jwst-crds.stsci.edu/ <https://jwst-crds.stsci.edu/>`_
+
+
+CRDS Servers
```
[the remark this answers is not in the document]
Comment (cshanahan1) [reply docs/jwst/user_documentation/reference_files_crds.rst:101]: i am incorporating these changes into my new documents

File: docs/jwst/user_documentation/running_pipeline_python.rst (hunk 11 above)
Comment (cshanahan1) [reply docs/jwst/user_documentation/running_pipeline_python.rst:504]: yeah thats what i was trying to describe, ill reword it.

**Comment by cshanahan1** [comment]
I addressed all the review comments, and rebased to add in the changes in #7342.

**Review by tapastro — APPROVED** [review]
Nice improvement in readability and organization!

**Comment by cshanahan1** [comment]
any objections to merging?

**Comment by cshanahan1** [comment]
ok, merging then. since this is a lot of text, if there are any issues you spot let me know and i will fix it!

**Comment by nden** [comment]
🎉

**Pull request merged**